src/internal/kopia/upload.go

```diff
@@ -1122,26 +1122,41 @@ func traverseBaseDir(
 	var hasItems bool
 
 	if changed {
-		err = fs.IterateEntries(
-			ctx,
-			dir,
-			func(innerCtx context.Context, entry fs.Entry) error {
-				dEntry, ok := entry.(fs.Directory)
-				if !ok {
-					hasItems = true
-					return nil
-				}
+		iter, err := dir.Iterate(ctx)
+		if err != nil {
+			return clues.WrapWC(ctx, err, "getting directory iterator")
+		}
+
+		// Need to keep err for the check after the loop as well so we also need to
+		// declare entry.
```

Comment:
oh, it was mostly because it's a bit odd to declare something when it's not used outside the scope of the for-loop. I'll move the comment down since it's a bit important